lib/ast-input-transform.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const attributeToPropertyMap = {
+  role: 'ariaRole',
+};
+
+class AngleBracketLinkToPolyfill {
+  transform(ast) {
+    let b = this.syntax.builders;
+
+    // in order to debug in https://https://astexplorer.net/#/gist/0590eb883edfcd163b183514df4cc717
+    // **** copy from here ****
+    function transformAttributeValue(attributeValue) {
+      switch (attributeValue.type) {
+        case 'TextNode':
+          return b.string(attributeValue.chars);
```

Comment:
Let’s propagate the original location info in these branches (so downstream error messages can use it)

Reply:
It seems I cannot do that with `b.string()`, can I? 

https://github.com/glimmerjs/glimmer-vm/blob/master/packages/%40glimmer/syntax/lib/builders.ts#L564

Btw, as AST docs are quite sparse/non-existent, what does this actually do and how is this used? For error messages to print concrete line/column numbers of hbs files that break something?

Reply:
>It seems I cannot do that with b.string(), can I?

Ugg, indeed you are correct. Though I think you can create it and then add the loc:

```js
let newString = b.string(attributeValue.chars);
newString.loc = attributeValue.loc;
return newString;
```

https://github.com/glimmerjs/glimmer-vm/blob/83a0c2fdfa8f22fd353dcab32ea41c4bcacaa379/packages/%40glimmer/syntax/lib/builders.ts#L577-L581 needs to be updated to take `loc` as the last arg in the returned function.

> what does this actually do and how is this used? For error messages to print concrete line/column numbers of hbs files that break something?

Right, when creating new AST nodes we should always attempt to propagate the original location information through to the new nodes. This helps to ensure that any error messages include the correct module name, line, and column number.